Timeout on bleeding

```
FAILED tests/everest/test_everserver.py::test_status_contains_max_runtime_failure - RuntimeError: Failed to get reply from server within configured timeout.
FAILED tests/everest/test_everserver.py::test_status_max_batch_num - RuntimeError: Failed to get reply from server within configured timeout.
FAILED tests/everest/test_detached.py::test_https_requests - RuntimeError: Failed to get reply from server within configured timeout.
```



https://github.com/equinor/komodo-releases/actions/runs/13104363883/job/36557254944

More failures added over last couple of days:
```
FAILED tests/everest/test_logging.py::test_logging_setup - RuntimeError: Failed to get reply from server within configured timeout.
FAILED tests/everest/test_everserver.py::test_status_max_batch_num - RuntimeError: Failed to get reply from server within configured timeout.
```

--- a/src/everest/detached/jobs/everserver.py
+++ b/src/everest/detached/jobs/everserver.py
@@ -3,6 +3,7 @@ import datetime
 import json
 import logging
 import os
+import random
 import socket
 import ssl
 import threading
@@ -273,7 +274,15 @@ def _everserver_thread(shared_data, server_config) -> None:
 
 
 def _find_open_port(host, lower, upper) -> int:
-    for port in range(lower, upper):
+    # Making the port selection random does not fix the problem that an
+    # everserver might be assigned a port that another everserver in the process
+    # of shutting down already have.
+    #
+    # Since this problem is very unlikely in the normal usage of everest this change
+    # is mainly for alowing testing to run in paralell.
+
+    for _ in range(10):
+        port = random.randint(lower, upper)
         try:
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             sock.bind((host, port))
@@ -283,7 +292,7 @@ def _find_open_port(host, lower, upper) -> int:
             logging.getLogger("everserver").info(
                 f"Port {port} for host {host} is taken"
             )
-    msg = f"No open port for host {host} in the range {lower}-{upper}"
+    msg = f"Failed 10 times to get a random port in the range {lower}-{upper} on {host}. Giving up."
     logging.getLogger("everserver").exception(msg)
     raise Exception(msg)
 

